index.js

```diff
@@ -142,57 +94,77 @@ module.exports = function (opts) {
       //but if not, set a short default (as needed in the tests)
       timeout_inactivity = timeout_inactivity || (opts.timers ? 600e3 : 5e3)
 
-      //set all timeouts to one setting, needed in the tests.
-      if(opts.timeout)
-        timeout_handshake = timeout_inactivity = opts.timeout
+      // needed for tests
+      if (!opts.connections)
+        opts.connections = {
+          incoming: {
+            net: [{ scope: "public", "transform": "shs" }]
+          },
+          outgoing: {
+            net: [{ transform: "shs" }]
+          }
+        }
+
+      var peers = api.peers = {}
 
-      var shsCap = (opts.caps && opts.caps.shs) || opts.appKey || appKey
-      var shs = Shs({
-        keys: opts.keys && toSodiumKeys(opts.keys),
-        seed: opts.seed,
-        appKey: toBuffer(shsCap),
+      var transports = []
 
-        //****************************************
-        timeout: timeout_handshake,
+      var transforms = [
+        //function () { return shs }
+      ]
 
-        authenticate: function (pub, cb) {
-          var id = '@'+u.toId(pub)
-          api.auth(id, function (err, auth) {
-            if(err) cb(err)
-            else    cb(null, auth || create.permissions.anonymous)
+      var server, ms, ms_client
+
+      function setupMultiserver () {
+        if(server) return server
+        if(transforms.length < 1)
+          throw new Error('secret-stack needs at least 1 transform protocol')
+
+        var server_suites = []
+        var client_suites = []
+
+        for (var incTransportType in opts.connections.incoming) {
+          opts.connections.incoming[incTransportType].forEach(function (conf) {
+            transforms.forEach(function (transform) {
+              transports.forEach(function (transport) {
+                if (transport.name == incTransportType && transform.name == conf.transform)
+                  server_suites.push([
+                    transport.create(conf),
+                    transform.create()
+                  ])
+              })
+            })
```

Comment:
@cryptix Here's the other one I mentioned. We can either do it here or in a separate PR, either way works great for me. Also instead of specifying `cartesian` and such I'm sure there's a library we can use. Maybe [lodash.product](https://www.npmjs.com/package/lodash.product)?